compiler/semantic/op.go

```diff
@@ -165,7 +165,7 @@ func semPool(ctx context.Context, scope *Scope, p *ast.Pool, ds *data.Source, he
 	}
 	var sources []dag.Op
 	for _, name := range poolNames {
-		source, err := semPoolWithName(ctx, scope, p, name, ds, head)
+		source, err := semPoolWithName(c.ctx, scope, p, name, c.source, c.head)
```

Comment:
Make this a checker method so you don't have to pass c.ctx, c.source, and c.head?